src/preview.js

```diff
@@ -1,51 +1,94 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
 import countries from '../assets/countries.json';
 import continentNames from '../assets/continent-names.json';
 import continents from '../assets/continents.json';
 import { getEmojiFlag } from './utils';
-import { __, sprintf } from '@wordpress/i18n';
+
+const textDomain = 'xwp-country-card';
 
 export default function Preview( { countryCode, relatedPosts } ) {
 	if ( ! countryCode ) return null;
 
 	const emojiFlag = getEmojiFlag( countryCode ),
-	      hasRelatedPosts = relatedPosts?.length > 0;
+		hasRelatedPosts = relatedPosts?.length > 0;
+
+	const helloTranslatedString = createInterpolateElement(
+		sprintf(
+			/* translators: country name, then country code, then continent name */
+			__(
+				'Hello from <strong>%1$s</strong> (<span>%2$s</span>), %3$s!',
+				textDomain
+			),
+			countries[ countryCode ],
+			countryCode,
+			continentNames[ continents[ countryCode ] ]
+		),
+		{
+			strong: <strong />,
+			span: <span className="xwp-country-card__country-code" />,
+		}
+	);
+
+	const displayRelatedPostsCount = () => {
+		if ( ! hasRelatedPosts ) {
+			return __( 'There are no related posts.', textDomain );
+		}
+		if ( relatedPosts.length === 1 ) {
+			return __( 'There is one related post:', textDomain );
+		}
+		return sprintf(
+			/* translators: %d is replaced with the number of related posts */
+			__( 'There are %d related posts:', textDomain ),
+			relatedPosts.length
+		);
+	};
+
+	const escape = ( dangerousString ) => {
+		const div = document.createElement( 'div' );
+		div.innerHTML = dangerousString;
+		return div.innerText;
+	};
 
 	return (
 		<div className="xwp-country-card">
-			<div className="xwp-country-card__media" data-emoji-flag={ emojiFlag }>
-				<div className="xwp-country-card-flag">
-					{ emojiFlag }
-				</div>
+			<div
+				className="xwp-country-card__media"
+				data-emoji-flag={ emojiFlag }
+			>
+				<div className="xwp-country-card-flag">{ emojiFlag }</div>
 			</div>
 			<h3 className="xwp-country-card__heading">
```

Comment:
There's a triple h3 in this component, which is a semantic issue.
It's also hard to guess if a h-type heading is appropriate in a component. Depending on where it's used, it might break the heading hierarchy. 
To be safe I'd replace them by spans or paragraphs.